File: Partner-Api/src/main/java/work/slhaf/partner/api/factory/entity/AgentBaseFactory.java
package work.slhaf.partner.api.factory.entity;

import work.slhaf.partner.api.factory.capability.exception.CapabilityFactoryExecuteFailedException;

import java.lang.reflect.InvocationTargetException;

public abstract class AgentBaseFactory {
    public void execute(AgentRegisterContext context) {
        try {
            setVariables(context);
            run();
        } catch (Exception e) {
            throw new CapabilityFactoryExecuteFailedException(e.getMessage(), e);
        }
    }

    protected abstract void setVariables(AgentRegisterContext context);

    protected abstract void run() throws NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException;
}

File: Partner-Framework/src/main/java/work/slhaf/partner/api/agent/factory/AgentBaseFactory.kt
package work.slhaf.partner.api.agent.factory

import work.slhaf.partner.api.agent.factory.context.AgentRegisterContext

/**
 * 所有注册链工厂的统一抽象。
 *
 * 每个工厂接收同一个 [AgentRegisterContext]，
 * 在其上完成单一阶段的扫描、校验、构建或注入。
 */
abstract class AgentBaseFactory {
    abstract fun execute(context: AgentRegisterContext)
}
